keras_cv/models/resnet_v1.py

```diff
@@ -0,0 +1,360 @@
+        return x
+
+    return apply
+
+
+def ResNet(
+    stackwise_filters,
+    stackwise_blocks,
+    stackwise_strides,
+    include_rescaling,
+    name="ResNet",
+    include_top=None,
+    weights=None,
+    input_shape=(None, None, 3),
+    pooling=None,
+    num_classes=None,
+    classifier_activation="softmax",
+    **kwargs,
+):
+    """Instantiates the ResNet architecture.
+
+    Args:
+      stackwise_filters: number of filters for each stack in the model.
+      stackwise_blocks: number of blocks for each stack in the model.
+      stackwise_strides: stride for each stack in the model.
+      include_rescaling: whether or not to Rescale the inputs. If set to True,
+        inputs will be passed through a `Rescaling(1/255.0)` layer.
+      name: string, model name.
+      include_top: whether to include the fully-connected
+        layer at the top of the network.
+      weights: one of `None` (random initialization),
+        or the path to the weights file to be loaded.
+      input_shape: optional shape tuple, defaults to (None, None, 3).
+      pooling: optional pooling mode for feature extraction
+        when `include_top` is `False`.
+        - `None` means that the output of the model will be
+            the 4D tensor output of the
+            last convolutional layer.
+        - `avg` means that global average pooling
+            will be applied to the output of the
+            last convolutional layer, and thus
+            the output of the model will be a 2D tensor.
+        - `max` means that global max pooling will
+            be applied.
+      num_classes: optional number of classes to classify images
+        into, only to be specified if `include_top` is True, and
+        if no `weights` argument is specified.
+      classifier_activation: A `str` or callable. The activation function to use
+        on the "top" layer. Ignored unless `include_top=True`. Set
+        `classifier_activation=None` to return the logits of the "top" layer.
+        When loading pretrained weights, `classifier_activation` can only
+        be `None` or `"softmax"`.
+      **kwargs: Pass-through keyword arguments to `tf.keras.Model`.
+
+    Returns:
+      A `keras.Model` instance.
+    """
+    if weights and not tf.io.gfile.exists(weights):
+        raise ValueError(
+            "The `weights` argument should be either `None` or the path to the "
+            "weights file to be loaded. Weights file not found at location: {weights}"
+        )
+
+    if include_top and not num_classes:
+        raise ValueError(
+            "If `include_top` is True, you should specify `num_classes`. "
+            f"Received: num_classes={num_classes}"
+        )
+
+    if include_top and pooling:
+        raise ValueError(
+            f"`pooling` must be `None` when `include_top=True`."
+            f"Received pooling={pooling} and include_top={include_top}. "
+        )
+
+    img_input = layers.Input(shape=input_shape)
```

Comment:
my recommendation would be to instead of reuse img_input, instead do:

img_input = layers.Input()
x = img_input

if include_rescaling:
    x = layers.Rescaling(1/255.0)(x)

x = layers.Conv2D(x)

This is less error prone than the approach you use here.